custom_components/jablotron_cloud/config_flow.py

```diff
@@ -1,73 +1,131 @@
 """Config flow for Jablotron Cloud integration."""
-from __future__ import annotations
 
 import logging
-from typing import Any
 
-from jablotronpy import Jablotron, UnexpectedResponse
 import voluptuous as vol
-
 from homeassistant import config_entries
-from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
-from homeassistant.core import HomeAssistant
-from homeassistant.data_entry_flow import FlowResult
-from homeassistant.exceptions import HomeAssistantError
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD, CONF_PIN, CONF_FORCE_UPDATE, CONF_SCAN_INTERVAL, \
+    CONF_TIMEOUT
+from jablotronpy import UnauthorizedException
 
 from .const import DOMAIN
+from .jablotron import JablotronClient
 
 _LOGGER = logging.getLogger(__name__)
 
-STEP_USER_DATA_SCHEMA = vol.Schema(
-    {
-        vol.Required(CONF_USERNAME): str,
-        vol.Required(CONF_PASSWORD): str,
-    }
-)
 
+def get_schema(config: dict) -> vol.Schema:
+    """Return config flow schema."""
+
+    return vol.Schema(
+        {
+            vol.Required(CONF_USERNAME, default=config.get(CONF_USERNAME, "")): str,
+            vol.Required(CONF_PASSWORD): str,
+            vol.Optional(CONF_PIN, default=config.get(CONF_PIN, "")): str,
+            vol.Optional(CONF_FORCE_UPDATE, default=config.get(CONF_FORCE_UPDATE, True)): bool,
+            vol.Optional(CONF_SCAN_INTERVAL, default=config.get(CONF_SCAN_INTERVAL, 30)): int,
+            vol.Optional(CONF_TIMEOUT, default=config.get(CONF_TIMEOUT, 15)): int
```

Comment:
Also do not like to have this settings. Why do you think we need it?

1. Tt exists only because jablotronpy was no reliable. It should have been always responsibility of jablotronpy to return reliable data in time or exception.
2. User has no idea why the number should be and what it means for him.
3. I was hopping you get rid of this completely by updating jablotronpy.

Reply:
This is not a JablotronPy problem, but a Jablotron API problem. Unfortunately, often the request takes a long time or gets stuck in a state where it keeps an open connection to the Cloud.
Even now, unfortunately, this happens and so I think it is useful to have a limit on when the API should get the data. What happens now is that if the limit is reached, at least the data that was able to be retrieved is applied - which is new.

And as I wrote above, the user sees some defaults and if he doesn't understand them, he shouldn't modify them. By default, the data is updated every 30 seconds and has 15 seconds to update.